app/services/change_schedule.rb

```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+
+class ChangeSchedule
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :cpd_lead_provider
+  attribute :participant_id
+  attribute :reason
+  attribute :course_identifier
+  attribute :schedule_identifier
+  attribute :cohort
+
+  alias_attribute :user_profile, :participant_profile
+  delegate :participant_profile_state, to: :participant_profile, allow_nil: true
+  delegate :lead_provider, to: :cpd_lead_provider, allow_nil: true
+
+  validates :course_identifier, course: true, presence: { message: I18n.t(:missing_course_identifier) }
+  validates :participant_id, presence: { message: I18n.t(:missing_participant_id) }
```

Comment:
I'm not sure we need the next 3 validations, if we add the induction record validator, let me know your thoughts